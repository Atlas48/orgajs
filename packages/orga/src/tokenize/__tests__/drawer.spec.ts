import tok from './tok'

describe('tokenize drawer', () => {
  it('knows drawer begins', () => {
    expect(tok(':PROPERTIES:')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": ":PROPERTIES:",
          "name": "PROPERTIES",
          "type": "drawer.begin",
        },
      ]
    `)
    expect(tok('  :properties:')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": ":properties:",
          "name": "properties",
          "type": "drawer.begin",
        },
      ]
    `)
    expect(tok('  :properties:  ')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": ":properties:",
          "name": "properties",
          "type": "drawer.begin",
        },
      ]
    `)
    expect(tok('  :prop_erties:  ')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": ":prop_erties:",
          "name": "prop_erties",
          "type": "drawer.begin",
        },
      ]
    `)
  })

  it('knows these are not drawer begins', () => {
    expect(tok('PROPERTIES:')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "PROPERTIES:",
          "type": "text.plain",
          "value": "PROPERTIES:",
        },
      ]
    `)
    expect(tok(':PROPERTIES')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": ":PROPERTIES",
          "type": "text.plain",
          "value": ":PROPERTIES",
        },
      ]
    `)
    expect(tok(':PR OPERTIES:')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": ":PR OPERTIES:",
          "type": "text.plain",
          "value": ":PR OPERTIES:",
        },
      ]
    `)
  })

  it('knows drawer ends', () => {
    expect(tok(':END:')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": ":END:",
          "type": "drawer.end",
        },
      ]
    `)
    expect(tok('  :end:')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": ":end:",
          "type": "drawer.end",
        },
      ]
    `)
    expect(tok('  :end:  ')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": ":end:",
          "type": "drawer.end",
        },
      ]
    `)
    expect(tok('  :end:  ')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": ":end:",
          "type": "drawer.end",
        },
      ]
    `)
  })

  it('knows these are not drawer ends', () => {
    expect(tok('END:')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "END:",
          "type": "text.plain",
          "value": "END:",
        },
      ]
    `)
    expect(tok(':END')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": ":END",
          "type": "text.plain",
          "value": ":END",
        },
      ]
    `)
    expect(tok(':ENDed')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": ":ENDed",
          "type": "text.plain",
          "value": ":ENDed",
        },
      ]
    `)
  })
})
