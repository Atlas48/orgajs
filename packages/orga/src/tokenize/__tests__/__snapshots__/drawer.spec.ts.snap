// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenize drawer knows drawer begins 1`] = `
Array [
  Object {
    "name": "PROPERTIES",
    "position": Object {
      "end": Object {
        "column": 12,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "drawer.begin",
  },
]
`;

exports[`tokenize drawer knows drawer begins 2`] = `
Array [
  Object {
    "name": "properties",
    "position": Object {
      "end": Object {
        "column": 14,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "drawer.begin",
  },
]
`;

exports[`tokenize drawer knows drawer begins 3`] = `
Array [
  Object {
    "name": "properties",
    "position": Object {
      "end": Object {
        "column": 14,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "drawer.begin",
  },
]
`;

exports[`tokenize drawer knows drawer begins 4`] = `
Array [
  Object {
    "name": "prop_erties",
    "position": Object {
      "end": Object {
        "column": 15,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "drawer.begin",
  },
]
`;

exports[`tokenize drawer knows drawer ends 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 5,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "drawer.end",
  },
]
`;

exports[`tokenize drawer knows drawer ends 2`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 7,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "drawer.end",
  },
]
`;

exports[`tokenize drawer knows drawer ends 3`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 7,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "drawer.end",
  },
]
`;

exports[`tokenize drawer knows drawer ends 4`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 7,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "drawer.end",
  },
]
`;

exports[`tokenize drawer knows these are not drawer begins 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 11,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "PROPERTIES:",
  },
]
`;

exports[`tokenize drawer knows these are not drawer begins 2`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 11,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": ":PROPERTIES",
  },
]
`;

exports[`tokenize drawer knows these are not drawer begins 3`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 13,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": ":PR OPERTIES:",
  },
]
`;

exports[`tokenize drawer knows these are not drawer ends 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 4,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "END:",
  },
]
`;

exports[`tokenize drawer knows these are not drawer ends 2`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 4,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": ":END",
  },
]
`;

exports[`tokenize drawer knows these are not drawer ends 3`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 6,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": ":ENDed",
  },
]
`;
